### PR TITLE
Introduced `regexp` option in String sanitizer

### PR DESCRIPTION
**Pull request by maciejkrol**
- Introduced `regexp` option in String sanitizer.
- Added specs.

### CODE REVIEW

File: lib/input_sanitizer/v2/types.rb [hunk 1]
```diff
@@ -44,6 +44,8 @@ def call(value, options = {})
         raise InputSanitizer::ValueNotAllowedError.new(value)
       elsif value.blank? && (options[:allow_blank] == false || options[:required] == true)
         raise InputSanitizer::BlankValueError
+      elsif options[:regexp] && Regexp.new(options[:regexp]).match(value) == nil
```
Comment (szymzet) [inline lib/input_sanitizer/v2/types.rb]: I would rather expect that `options[:regexp]` is already an object of type `Regexp` and not create a new one (and thus making passing a raw string like `"[a-b]"` a valid input for regexp sanitizer.

`options[:regexp].match(value).nil?` should do the trick

File: lib/input_sanitizer/errors.rb [hunk 2]
```diff
@@ -30,6 +30,16 @@ def initialize
     end
   end
 
+  class RegexpMismatchError < ValidationError
+    def code
+      :regexp
+    end
+
+    def initialize(regexp)
+      super("does not match the regular expression: #{regexp}")
```
Comment (szymzet) [inline lib/input_sanitizer/errors.rb]: I am not 100% sold on the fact of exposing our validation regexp to the client. IMO this is too much of an implementation detail that we use a regexp. I think this is case by case, e.g. we could use regexp for validating UUID format and then the error message sent to the user should be "not a valid UUID".

What is your regexp use case? I'm wondering wheter you have something that exposing the regexp to the user of th API would make sens.

File: lib/input_sanitizer/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module InputSanitizer
-  VERSION = "0.3.27"
```
Comment (szymzet) [inline lib/input_sanitizer/version.rb]: bump version in other commit/pr - our current best practice is to add features and then release the features in a saparate step

File: lib/input_sanitizer/errors.rb (hunk 2 above)
Comment (maciejkrol) [reply lib/input_sanitizer/errors.rb]: I am not sure who do you consider a _client_ in this sense. This is a gem so I assume that the client is the application using it. These errors should not be exposed further.

In my use case, I validate the input in a controller and if it's invalid return HTTP/422 without any details. We don't want to give a hint of what is wrong in this case. Even if we wanted, still I think that we should return something that makes sense in the context of the service not simply return validation errors to the client.

What do you think?

Comment (szymzet) [reply lib/input_sanitizer/errors.rb]: Perhaps I'm mistaken but I think the errors are often forewarded in responses as-is. As we agreed offline a good compromise would be not exposing the regexp cause in practice the use of the gem should tweak the message to suite the business case (e.g. "uuid invalid" in the mentioned example)

Comment (maciejkrol) [reply lib/input_sanitizer/errors.rb]: Sure, let's not expose the regexp.

**Pull request merged**